frontend/src/Authentication/MsalService.ts

```diff
@@ -0,0 +1,64 @@
+import * as msal from "@azure/msal-browser";
+import AccountDetails from "../DataClasses/AccountDetails";
+
+const msalConfig = {
+    auth: {
+        clientId: 'e1234a8c-f3d3-4bec-ae16-7168aeafd19c',
+        authority: 'https://partyklinest.b2clogin.com/partyklinest.onmicrosoft.com/B2C_1_PARTYKLINEST_USER_FLOW',
+        knownAuthorities: ['https://partyklinest.b2clogin.com'],
+        redirectUri: 'http://localhost:3000/'
+    }
+};
+
+const msalInstance = new msal.PublicClientApplication(msalConfig);
+
+export const RetrieveToken = () => {
+    return msalInstance.acquireTokenSilent({scopes:[]}).then((tokenResponse) => {
+        if (tokenResponse === null) return undefined;
+        msalInstance.setActiveAccount(tokenResponse.account);
+        return tokenResponse.accessToken;
+    }).catch((error) => {
+        console.log(error);
+        return undefined;
+    })
+}
+
+export const B2CLogin = async () => {
+    try {
+        const loginResponse = await msalInstance.loginPopup({scopes:[]});
+        if (loginResponse === null) return undefined;
+        msalInstance.setActiveAccount(loginResponse.account);
+        return loginResponse.accessToken;
+    } catch (err) {
+        // handle error
+        console.log(err);
+        return undefined;
+    }
+}
+
+export const B2CLogout = () => msalInstance.logoutPopup();
+
+export const B2CEditProfile = async () => {
+    try {
+        const loginResponse = await msalInstance.loginPopup({scopes:[], authority:'https://partyklinest.b2clogin.com/partyklinest.onmicrosoft.com/B2C_1_EDIT_PROFILE'});
```

Comment:
Do env

Reply:
@szymon-zygula specyfikacja jest niezgodna ze specyfikacją. Wrzucili specyfikację, jakie pola mają być podane w formsie na Azure: https://github.com/maciejBart99/partyKLINer_docs/blob/main/Atrybuty.png i https://github.com/maciejBart99/partyKLINer_docs/blob/main/Oswiadczenia.png i według tego jest zrobione.

Reply:
Potwierdzam, robię to w taki sposób, jak dane wyspecyfikowane są w Azure